plugin/loader/loader.go

```diff
@@ -83,16 +86,32 @@ type PluginLoader struct {
 	state   loaderState
 	plugins map[string]plugin.Plugin
 	started []plugin.Plugin
+	config  config.Plugins
+	repo    string
 }
 
 // NewPluginLoader creates new plugin loader
-func NewPluginLoader() (*PluginLoader, error) {
-	loader := &PluginLoader{plugins: make(map[string]plugin.Plugin, len(preloadPlugins))}
+func NewPluginLoader(repo string) (*PluginLoader, error) {
+	loader := &PluginLoader{plugins: make(map[string]plugin.Plugin, len(preloadPlugins)), repo: repo}
+	if repo != "" {
+		cfg, err := cserialize.Load(filepath.Join(repo, config.DefaultConfigFile))
+		switch err {
+		case cserialize.ErrNotInitialized:
+		case nil:
+			loader.config = cfg.Plugins
+		default:
+			return nil, err
+		}
+	}
 	for _, v := range preloadPlugins {
 		if err := loader.Load(v); err != nil {
 			return nil, err
 		}
 	}
+
+	if err := loader.LoadDirectory(filepath.Join(repo, "plugins")); err != nil {
```

Comment:
Abs patch 2/2
```suggestion
	if err = loader.LoadDirectory(filepath.Join(repo, "plugins")); err != nil {
```

Reply:
In general, I don't like using `=` in assignments inside if statements. The point of using an if statement like this is to scope the `err` variable to the if statement. Using a normal assignment lets it escape the scope and can be really confusing.

Reply:
Basically, this should either be left as-is or should be:

```
err = loader.LoadDirectory(...)
if err != nil {
  // ...
}
```